client/login/wp-login/index.jsx

```diff
@@ -104,12 +118,16 @@ class Login extends React.Component {
 				</a>;
 		}
 
-		const goBackLink = ! magicLoginView && <a
-			href="#"
-			key="back-link"
-			onClick={ this.goBack }>
-				<Gridicon icon="arrow-left" size={ 18 } /> { this.props.translate( 'Return' ) }
-			</a>;
+		let goBackLink;
+		if ( this.state.loaded && window.history.length > 1 ) {
```

Comment:
My bad, it's indeed possible to land on this page by following a link, meaning the history wouldn't be empty. Not sure why my brain couldn't envision such frequent scenario ...